RocketChat/Rocket.Chat
# [IMPROVE] Read receipts in threads

## Proposed changes (including videos or screenshots)


- Create new `afterReadThread` callback to handle read receipts for messages in threads;
- Add read receipts for messages in threads only when the thread is open;
- Remove the read/unread checkmark for messages in threads.



## Issue(s)


## Steps to test or reproduce

1. Go to **Administration > Message** and enable "Show Read Receipts" and "Detailed Read Receipts";
2. Login with two users and access the same room with both of them;
3. As "User 1", send a message in an existing thread;
4. Access this room as "User 2". **_Expected behavior:_** the message sent in step 3 should _NOT_ be marked as read by "User 2";
5. As "User 2", send a message in the room (not in the thread). **_Expected behavior:_** the message sent in step 3 should still _NOT_ be marked as read by "User 2";
6. Open the thread as "User 2". **_Expected behavior:_** the message sent in step 3 should be marked as read by "User 2";
7.  Send a new message in the thread. **_Expected behavior:_** the new message should be instantly marked as read by both users.

## Further comments
